analytics/src/test/java/com/segment/analytics/AnalyticsBuilderTest.kt

```diff
@@ -0,0 +1,280 @@
+package com.segment.analytics
+
+import android.Manifest.permission.INTERNET
+import android.app.Application
+import android.content.Context
+import android.content.pm.PackageManager.PERMISSION_DENIED
+import android.content.res.Resources
+import com.segment.analytics.Analytics.Builder
+import com.segment.analytics.Analytics.WRITE_KEY_RESOURCE_IDENTIFIER
+import com.segment.analytics.core.BuildConfig
+import org.assertj.core.api.Assertions
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Matchers.eq
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import java.util.concurrent.TimeUnit
+
+@RunWith(RobolectricTestRunner::class)
+@Config(constants = BuildConfig::class, sdk = [18], manifest = Config.NONE)
+class AnalyticsBuilderTest {
+
+    lateinit var context: Application
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        context = TestUtils.mockApplication()
+        Analytics.INSTANCES.clear()
+        Mockito.`when`(context.applicationContext).thenReturn(context)
```

Comment:
why is the `when` wrapped in backticks?

Reply:
oh i see! its cos `when` is a keyword in kotlin.
MIght wanna try this out then
https://www.baeldung.com/kotlin-mockito 
```
library exposes a new whenever() method that can be used freely, without the need for back-ticks like we had to when using Mockito's native when() method.
```